### ISSUE TITLE
Change request: Set msg.output

### ISSUE THREAD

**Issue by hubertguillemain** [bartbutenaers/node-red-contrib-msg-router#3]
The router provides the ability to reconfigure the outputs on the fly which is great.
The problem I have is that to identidy which port to reconfigure if my Nth subflow after the router node fails, I have to use a Change node with 1 line to set the port.

It would be helpful that all output messages of the router node have the porpery msg.output set to the port number which outputed the message

**Comment by hubertguillemain** [comment]
To give some more insight on what I'm doing:
I'm pooling Twitter search nodes to overcome the quota limitation, based on sequential round robin.
If a Twitter search node's account exceeds the rate limit, I want to de-activate it's output in the router, and reactivate it 15 minutes later

![image](https://user-images.githubusercontent.com/30046324/47783861-1e22b980-dd04-11e8-8abd-2d9d0a8b6757.png)

**Comment by bartbutenaers** [comment]
Hi Hubert,

That indeed makes sense.  Will try to add it as soon as possible, and will keep you updated.  

Kind regards,
Bart

**Comment by bartbutenaers** [comment]
Hi Hubert, 

I have updated the version on Github.  Would be nice if you could test whether it fits your needs, before I publish it on NPM.  You can install it directly from Github like this:
```
npm install bartbutenaers/node-red-contrib-msg-router
```
For example here is how it looks like using Round Robin routing:

![image](https://user-images.githubusercontent.com/14224149/47814044-f99eff80-dd4c-11e8-9ba4-5c536667b2d4.png)

Some remarks:
+ I have used **1-based** port numbers (i.e. first port is 1, not 0) since the first port in the config screen is also 1.  Moreover in case of message based routing, the msg.output (in the input message) should be also 1 for the first output.
+ The msg.output will not be available in case of broadcast routing.  Indeed in this case the SAME message will be send to ALL outputs, if the cloning checkbox is deselected.
+ The help text in the info panel has also been updated:
![image](https://user-images.githubusercontent.com/14224149/47814535-446d4700-dd4e-11e8-996a-8e1cf7b34025.png)

+ The readme page on Github has also been updated
![image](https://user-images.githubusercontent.com/14224149/47815107-b3976b00-dd4f-11e8-8976-e91fc201e307.png)

Hope to hear from you soon!
Bart

**Comment by hubertguillemain** [comment]
Hi Bart,

This looks perfect to me. Its a long week end starting here so I wont be able to actually clone the github version until Monday, but from what I read it's exactly as I would expect it.

Thanks of lot for great support !
Cheers,
Hubert

**Comment by hubertguillemain** [comment]
Getting some build errors - still Googleing for a solution..

C:\Users\hguillemain\.node-red\node_modules\ursa>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  ursaNative.cc
  win_delay_load_hook.cc
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(389): error C2027: utilisation du type non défini 'r
sa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(390): error C2027: utilisation du type non défini 'r
sa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(407): error C2027: utilisation du type non défini 'r
sa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(536): error C2027: utilisation du type non défini 'r
sa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(536): error C2660: 'bignumToBuffer' : la fonction ne
 prend pas 1 arguments [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(100): note: voir la déclaration de 'bignumToBuffer
  '
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(552): error C2027: utilisation du type non défini 'r
sa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(552): error C2660: 'bignumToBuffer' : la fonction ne
 prend pas 1 arguments [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(100): note: voir la déclaration de 'bignumToBuffer
  '
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(567): error C2027: utilisation du type non défini 'r
sa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(567): error C2660: 'bignumToBuffer' : la fonction ne
 prend pas 1 arguments [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(100): note: voir la déclaration de 'bignumToBuffer
  '
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1218): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1219): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1220): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1221): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1222): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1223): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1224): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1225): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1270): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
c:\users\hguillemain\.node-red\node_modules\ursa\src\ursanative.cc(1271): error C2027: utilisation du type non défini '
rsa_st' [C:\Users\hguillemain\.node-red\node_modules\ursa\build\ursaNative.vcxproj]
  c:\users\hguillemain\.node-gyp\10.11.0\include\node\openssl\evp.h(914): note: voir la déclaration de 'rsa_st'
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\hguillemain\.node-red\node_modules\ursa
gyp ERR! node -v v10.11.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ursa@0.9.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ursa@0.9.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\hguillemain\AppData\Roaming\npm-cache\_logs\2018-11-05T12_11_40_960Z-debug.log

**Comment by bartbutenaers** [comment]
Bonsoir Hubert,

It seems you are running Node-RED on Windows.  I think you have to install it like this (but not sure since I'm running Node-RED on Raspberry PI):
```
cd %HOMEPATH%\.nodered
npm install bartbutenaers/node-red-contrib-msg-router
```
Keep me informed!
Bart

**Comment by hubertguillemain** [comment]
This is actually what I was trying to do, but not working unfortunately :(
I'd say from you rdescription it looks as I'd expect

**Comment by bartbutenaers** [comment]
Hubert,
No problem!  I will test it again this weekend by myself, and publish it on npm afterwards.
Will keep you informed here...
Bart

**Comment by hubertguillemain** [comment]
FYI, I finally found the issue, which was relevant for other nodes: https://github.com/nodejs/node-gyp/issues/809
What eventually worked was the following line to force-reinstall node-gyp
npm install -g --unsafe-perm node-gyp

So, I could finally test. The node works as expected, although I can't see the indication in the info panel.

**Comment by bartbutenaers** [comment]
Hey Hubert,
Thank you for testing!

The indication in the info panel is the last section at the bottom:

![image](https://user-images.githubusercontent.com/14224149/48370097-969c5980-e6b8-11e8-82b1-d4b109ac50d2.png)

It is now available as version **0.0.4** in the palette:

![image](https://user-images.githubusercontent.com/14224149/48370130-b0d63780-e6b8-11e8-95cf-c2e97eb81e3e.png)

Good luck with it!
Bart
